src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -1481,7 +1481,12 @@ private BoundExpression BindDefaultExpression(DefaultExpressionSyntax node, Diag
                 expression = null;
                 if (node is IdentifierNameSyntax identifier)
                 {
-                    var type = BindNativeIntegerSymbolIfAny(identifier, diagnostics);
+                    // Don't bind nameof(nint) or nameof(nuint) so that ERR_NameNotInContext is reported.
+                    // The check for ArgumentSyntax is to allow the binding for something like nameof(nint.Equals).
+                    var type = IsInsideNameof && identifier.Parent is ArgumentSyntax
```

Comment:
Do I need to check that the parent of this ArgumentSyntax is an ArgumentListSyntax of an InvocationExpressionSyntax whose Expression is an IdentifierNameSyntax with IdentifierToken.Text equals to nameof?
That would be a large check that I don't know it's necessary or not.

For the above to make some sense, refer to the following visualization while reading it :smile:

![image](https://user-images.githubusercontent.com/31348972/92325148-8a7d4b00-f048-11ea-9e1d-c9c552549d94.png)